File: examples/math.rs
use vinculum_hs::functions::math::{add, factorial, multiply, testfn};

#[vinculum_hs::main]
fn main() {
    let a = 5;
    let b = 10;

    let result = add(a, b);
    println!("{a} + {b} = {result}");

    let result = multiply(a, b);
    println!("{a} * {b} = {result}");

    let result = factorial(a);
    println!("Factorial 5 = {result}");

    println!("{}", testfn(4));
}

File: examples/haskell/Math.hs
module Math where

import Data.Int

-- Add function
add :: Int64 -> Int64 -> Int64
add a b = a + b

multiply :: Int64 -> Int64 -> Int64
multiply a b = a * b

factorial :: Int64 -> Int64
factorial 0 = 1
factorial n = n * factorial (n - 1)

testfn :: a -> a
testfn arg = arg